model/PermissionsService.php

```diff
@@ -228,14 +328,12 @@ private function validateResources(array $resultPermissions): void
         }
     }
 
-    /**
-     * @param array $addRemove
-     * @param string $resourceId
-     * @param bool $isRecursive
-     */
-
-    private function triggerEvents(array $addRemove, string $resourceId, bool $isRecursive): void
-    {
+    private function triggerEvents(
+        array $addRemove,
+        string $resourceId,
+        bool $isRecursive,
+        bool $applyToNestedResources
+    ): void {
         if (!empty($addRemove['add'])) {
```

Comment:
hmmm, interesting, this function is called triggerEvents, meaning it SHALL trigger something :D . however i see a check prior to it !empty($addRemove['add'] and this means that there could be cases where no event is triggered. I wonder if this check could be taken out to the one calling this function :thinking:

Reply:
We will always trigger the last two events in this method no matter what (line 347, [here](https://github.com/oat-sa/extension-tao-dac-simple/blob/feat/ADF-1320/prototype-subtasks-acl/model/PermissionsService.php#L347)).

```
        $this->eventManager->trigger(
            new DataAccessControlChangedEvent(
                $resourceId,
                $addRemove,
                $isRecursive,
                $applyToNestedResources
            )
        );
```

The name is associated with the natural way of enumerating the steps that are carried out by `applyPermissions`, which I find easy to read:

1. `getResourcesToUpdate(...)`
2. `getResourcesPermissions(...)`
3. `getDeltaForResourceTree(...)`
4. `getActions(...)`
5. `dryRun($actions, $currentPermissions)`
6. `wetRun($actions)`
7. `triggerEvents(...)`

I'd prefer to keep this logic here (i.e. to check which events are to/are not to be triggered). BTW I'm open to suggestions for a new name :slightly_smiling_face: